sycl/doc/syclgraph/SYCLGraphUsageGuide.md

```diff
@@ -680,3 +680,64 @@ execMainGraph.update(updateGraph);
 // ptrA
 myQueue.ext_oneapi_graph(execMainGraph);
 ```
+
+### Graph-Owned Memory Allocations
```

Comment:
Might be worth having an example that uses a memory pool, if we're going to share this draft with library teas for feedback.

Reply:
I've updated one of the examples here to use a memory pool.